libs/datamodel/connectors/datamodel-connector/src/lib.rs

```diff
@@ -251,6 +255,14 @@ pub trait Connector: Send + Sync {
         NativeTypeErrorFactory::new(instance.to_string(), self.name().to_owned())
     }
 
+    fn allowed_index_operator_classes(
+        &self,
+        _algo: IndexAlgorithm,
+        _field: ScalarFieldWalker<'_>,
+    ) -> Vec<OperatorClass> {
+        Vec::new()
+    }
```

Comment:
I am really uncomfortable with these operator classes in core or the connector API. I'll try to find time this week to change that.

Reply:
And I'll have to change the API for these completions as part of that.

Reply:
maybe we can downcast the connector as an `Option<T>` and have the connector-specific things there?